src/figuresClasses.ts

```diff
@@ -1,19 +1,75 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'orange' | 'yellow' | 'green' | 'blue' | 'purple';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('sides must be > 0');
+    }
+
+    if ((a > b && a > c && a >= (b + c))
+      || (b > a && b > c && b >= (a + c))
+      || (c > a && c > b && c >= (a + b))) {
+      throw new Error(`sides ${a}, ${b} and ${c} can't form a triangle`);
+    }
+  }
 
+  getArea(): number {
+    const s: number = (this.a + this.b + this.c) / 2;
+
+    return +Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c))
+      .toFixed(2);
+  }
 }
 
 export class Circle {
+  shape: Shape = 'circle';
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('radius must be > 0');
+    }
+  }
 
+  getArea(): number {
```

Comment:
The area for circle is not being rounded down to hundredths. Use .toFixed(2) method.

Reply:
it is rounded, why not? not with .toFixed() method, but it's rounded

I had to use my decision because with .toFixed() didn't work one test (there were the result like 113.1 and test expected 113.09)